### PR TITLE
add a basic Gitlab org backend

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,4 +5,5 @@
 /.coverage
 /.tox
 /venv*
+.env*
```
Comment (asottile) [inline .gitignore]: unless this file is generated from the repo please rebase out this commit

File: tests/conftest.py [hunk 2]
```diff
@@ -78,6 +78,12 @@ def not_a_tty():
         yield mck
 
 
+@pytest.fixture(autouse=True)
+def locale_environ(monkeypatch):
+    # ensures git output is english
+    monkeypatch.setenv('LC_ALL', 'C')
```
Comment (asottile) [inline tests/conftest.py]: I prefer
```python
with mock.patch.dict(os.environ, {'LANG': 'C.UTF-8'}):
    yield
```

as the scope is more well-defined than `monkeypatch` and it is more nice to python and its unicode support (`LANG=C` will cause default-ascii in several places)

I'm curious in the actual failures that this fixes though and this patch should be split into another PR -- could you include the failures in the issue you created?

File: all_repos/github_api.py [hunk 3]
```diff
@@ -47,6 +47,18 @@ def get_all(url: str, **kwargs: Any) -> List[Dict[str, Any]]:
     return ret
 
 
+def filter_repos_from_settings(
+    repos: List[Dict[str, Any]], settings: Any,
```
Comment (asottile) [inline all_repos/github_api.py]: this makes the call not typesafe, which is why it wasn't moved in the first place

there's no reason to have both `filter_repos` and `filter_repos_from_settings` as well

I would just drop this commit

File: tests/conftest.py (hunk 2 above)
Comment (RonnyPfannschmidt) [reply tests/conftest.py]: sure, bascially the problem is that git output was german on mysystem, i'll follow up on it the next time i touch my work pc

Comment (asottile) [reply tests/conftest.py]: oh actually! even better this project uses `pytest-env` so that can be set there!

Comment (RonnyPfannschmidt) [reply tests/conftest.py]: moved to #110

File: all_repos/github_api.py (hunk 3 above)
Comment (RonnyPfannschmidt) [reply all_repos/github_api.py]: removed

**Comment by asottile** [comment]
looks good!  just needs some cleanup of the dead code and such but I think this is ready for merging!

**Review by asottile — APPROVED** [review]

**Pull request merged**